tests/testthat/test-metadata_to_coldata.R

```diff
@@ -0,0 +1,189 @@
+    colData(sce)[["sum"]] <- runif(ncol(sce))
+
+    # add some metadata
+    library_id <- batch
+    sample_id <- batch
+
+    metadata(sce)$library_id <- library_id
+    metadata(sce)$sample_id <- sample_id
+    metadata(sce)$sample_metadata <- data.frame(
+      sample_id = sample_id,
+      library_id = library_id
+    )
+
+    return(sce)
+  }
+)
+
+# create merged sce
+merged_sce <- merge_sce_list(sce_list = sce_list,
+                             retain_coldata_cols = "sum")
+
+
+test_that("`metadata_to_coldata` works as expected with a single object", {
+
+  sce_with_metadata <- metadata_to_coldata(sce,
+                                           join_columns = "library_id")
+
+  # check that colData now contains sample id and diagnosis columns
+  expect_contains(
+    colnames(colData(sce_with_metadata)),
+    c("sample_id", "diagnosis")
+  )
+
+  # check that sample id column contains expected value
+  expect_equal(
+    unique(colData(sce_with_metadata)$sample_id),
+    "sample1"
+  )
+
+})
+
+test_that("`metadata_to_coldata` works as expected joining on multiple columns", {
+
+  # add a second join column
+  sce$sample_id <- "sample1"
+
+  sce_with_metadata <- metadata_to_coldata(sce,
+                                           join_columns = c("library_id", "sample_id"))
+
+  # check that colData now contains sample id column
+  expect_contains(
+    colnames(colData(sce_with_metadata)),
+    c("sample_id", "diagnosis")
+  )
+
+  # check that sample id column contains expected value
+  expect_equal(
+    unique(colData(sce_with_metadata)$sample_id),
+    "sample1"
+  )
+
+
+})
+
+test_that("`metadata_to_coldata` works as expected with a merged object", {
+
+
+  merged_sce_with_metadata <- metadata_to_coldata(merged_sce,
+                                                  join_columns = "library_id")
+
+  # check that colData now contains sample id column
+  expect_contains(
+    colnames(colData(merged_sce_with_metadata)),
+    "sample_id"
+  )
+
+  # check that sample id column contains expected value
+  expect_setequal(
+    unique(colData(merged_sce_with_metadata)$sample_id),
+    c("sce1", "sce2", "sce3")
+  )
+
+})
+
+test_that("`metadata_to_coldata` fails as expected", {
+
+  # sce is required
+  expect_error(metadata_to_coldata(sce = "not an sce"))
+
+  # column name is missing from colData
+  expect_error(metadata_to_coldata(sce,
+                                   join_columns = "not a column"))
+
+  # column is missing from sample metadata
+  # add a new column that is only in the colData and not in the sample metadata
+  sce$batch_column <- "library1"
+  expect_error(metadata_to_coldata(sce,
+                                   join_columns = "batch_column"))
+
+  # no sample metadata present
+  metadata(sce) <- metadata(sce)[!names(metadata(sce)) %in% "sample_metadata"]
+  expect_warning(metadata_to_coldata(sce,
+                                   join_columns = "library_id"))
+
```

Comment:
I would move this to a different testing block, since this condition is not actually a failure!